pkg/planner/core/task.go

```diff
@@ -131,7 +134,7 @@ func (p *PhysicalUnionScan) Attach2Task(tasks ...base.Task) base.Task {
 			p.SetChildren(sel)
 			p.SetStats(task.Plan().StatsInfo())
 			rt, _ := task.(*RootTask)
```

Comment:
```suggestion
			rt := task.(*RootTask)
```

Or you can add the assert here.